modules/javafx.controls/src/main/java/javafx/scene/control/skin/PaginationSkin.java

```diff
@@ -194,20 +194,27 @@ public PaginationSkin(final Pagination control) {
         nextStackPane.getStyleClass().add("page");
         nextStackPane.setVisible(false);
 
-        resetIndexes(true);
-
         this.navigation = new NavigationControl();
 
         getChildren().addAll(currentStackPane, nextStackPane, navigation);
 
-        control.maxPageIndicatorCountProperty().addListener(o -> {
+        ListenerHelper.get(this).addInvalidationListener(getSkinnable().maxPageIndicatorCountProperty(), (o) -> {
```

Comment:
Minor: In the constructor, I recommend using the `control` argument rather than calling `getSkinnable()`, although this is just a recommendation.

Minor: since you make several calls to `ListenerHelper`, a temporary variable might make the code simpler; this is also just a recommendation.